core/src/test/java/org/apache/iceberg/TestTableMetadata.java

```diff
@@ -314,8 +319,11 @@ public void testBackwardCompat() throws Exception {
     assertThat(metadata.snapshot(previousSnapshotId).schemaId()).isNull();
   }
 
-  @Test
-  public void testInvalidMainBranch() throws IOException {
+  @ParameterizedTest
+  @FieldSource("org.apache.iceberg.TestHelpers#ALL_VERSIONS")
```

Comment:
Instead of this, we could also put it as: 
```suggestion
  @ValueSource(ints = MIN_FORMAT_VERSION_V2)
```

This avoids having to rely on the `assumeThat`, which I feel makes it harder to interpret the test. I'm not super strong on this, though. Maybe our test-connaisseur @nastra can share his opinion here.

Reply:
I'd suggest to do the same as mentioned in https://github.com/apache/iceberg/pull/11948#discussion_r1916254927 and parameterize this at the class level